auth-foundation/src/androidTest/java/com/okta/authfoundation/credential/SharedPreferencesTokenStorageTest.kt

```diff
@@ -156,6 +162,42 @@ class SharedPreferencesTokenStorageTest {
         }
     }
 
+    @Test fun testKeyGenParameterSpecIsUsed() {
+        val spec = KeyGenParameterSpec.Builder(
+            "com_okta_authfoundation_storage",
+            KeyProperties.PURPOSE_ENCRYPT or KeyProperties.PURPOSE_DECRYPT
+        )
+            .setBlockModes(KeyProperties.BLOCK_MODE_GCM)
+            .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_NONE)
+            .setUserAuthenticationRequired(true)
+            .setUserAuthenticationParameters(10, KeyProperties.AUTH_BIOMETRIC_STRONG)
+            .setKeySize(256)
+            .build()
+        val eventCoordinator = EventCoordinator(eventHandler)
+        subject = SharedPreferencesTokenStorage(
+            json = Json,
+            dispatcher = Dispatchers.Unconfined,
+            eventCoordinator = eventCoordinator,
+            context = ApplicationProvider.getApplicationContext(),
+            keyGenParameterSpec = spec,
+        )
+        val exception = assertFailsWith<Exception> {
+            runBlocking {
+                subject.add("element")
+            }
+        }
+        if (exception is InvalidAlgorithmParameterException) {
+            // This is the exception we expect if no biometrics are enrolled.
+            assertThat(exception.cause).hasMessageThat().isEqualTo("At least one biometric must be enrolled to create keys requiring user authentication for every use")
+        } else {
+            // This is the exception we expect if biometrics are enrolled.
+            assertThat(exception).isInstanceOf(KeyStoreException::class.java)
+            assertThat(exception).hasMessageThat().isEqualTo("the master key android-keystore://com_okta_authfoundation_storage exists but is unusable")
+            assertThat(exception.cause).isInstanceOf(UserNotAuthenticatedException::class.java)
+            assertThat(exception.cause).hasMessageThat().isEqualTo("User not authenticated")
+        }
```

Comment:
Having two different assert branches could cause us to silently miss coverage. Is it possible to set up both of these scenarios as separate tests?

Reply:
Good call out. If this were testing our code, I'd 100% agree with your conclusion, that we should test both scenarios.

What I'm really trying to test here is that we're using the passed in `KeyGenParameterSpec`, and not the default one for example.

If you have a better idea on how to test that we're using the supplied `KeyGenParameterSpec`, I'm definitely open to changing this.

Reply:
I was originally thinking of setting up the test case to add biometrics through UIAutomator. But, since this is a public API being used by us, that's a lot of effort for code which should be tested in Android's codebase. I'm fine with this approach considering that, and the test is doing what the test name says it should be doing.